contracts/src/gateway/GatewayManagerFacet.sol

```diff
@@ -171,14 +171,15 @@ contract GatewayManagerFacet is GatewayActorModifiers, ReentrancyGuard {
         supplySource.expect(SupplyKind.ERC20);
 
         // Lock the specified amount into custody.
-        supplySource.lock({value: amount});
+        // Account for tokens with Transfer Fees
+        uint256 transferAmount = supplySource.lock({value: amount});
```

Comment:
Let's document that if the transfer amount is zero, this will have reverted.

Reply:
Let's also document that this can return a higher effectively locked amount if the token is inflationary.

Reply:
521cb344